### PR TITLE
Avoid logging total time elapsed for tab command since it will remove the intellisense

### PR DESCRIPTION
**Pull request by jainaashish**
Avoid logging total time elapsed for tab command since it will remove the intellisense

Fix https://github.com/NuGet/Home/issues/2853

### CODE REVIEW

**Comment by jainaashish** [comment]
@alpaix @rrelyea @emgarten

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs [hunk 1]
```diff
@@ -88,11 +88,14 @@ protected NuGetPowerShellBaseCommand()
...
         {
-            get { return new NuGetPackageManager(
-                _sourceRepositoryProvider,
-                ConfigSettings,
-                VsSolutionManager,
-                _deleteOnRestartManager); }
+            get
+            {
+                return new NuGetPackageManager(
+              _sourceRepositoryProvider,
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs]: tabbing

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs [hunk 2]
```diff
@@ -218,7 +221,12 @@ protected override sealed void ProcessRecord()
             }
 
             stopWatch.Stop();
-            LogCore(ProjectManagement.MessageLevel.Info, string.Format(CultureInfo.CurrentCulture, Resources.Cmdlet_TotalTime, stopWatch.Elapsed));
+
+            // Log total time elapsed except for Tab command
+            if ((this as TabExpansionCommand) == null)
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs]: if (!(this is TabExpansionCommand)), better yet - have a member indicating if we should log timing or not, and avoid casting

**Comment by yishaigalatzer** [comment]
:watch:

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/TabExpansionCommand.cs [hunk 3]
```diff
@@ -16,6 +16,9 @@ public class TabExpansionCommand : FindPackageCommand
 
         protected override void ProcessRecordCore()
         {
+            // don't log total time elapsed for Tab command
+            IsLoggingTimeEnabled = false;
```
Comment (alpaix) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/TabExpansionCommand.cs]: Initialize it in the constructor.

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs [hunk 4]
```diff
@@ -201,6 +209,9 @@ protected override sealed void ProcessRecord()
             var stopWatch = new Stopwatch();
             stopWatch.Start();
 
+            // by default logging time is enabled
+            IsLoggingTimeEnabled = true;
```
Comment (alpaix) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs]: put the default value on property declaration line 202.

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs [hunk 5]
```diff
@@ -193,6 +196,11 @@ protected IErrorHandler ErrorHandler
             get { return this; }
         }
 
+        /// <summary>
+        /// Determine if needs to log total time elapsed or not
+        /// </summary>
+        protected bool IsLoggingTimeEnabled { get; set; }
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs]: This should be a virtual get only property, and then you never need to set the value. Make it `IsLoggingTimeDisabled` so its off by default and then you only need to return true in one implementation

**Comment by alpaix** [comment]
:shipit:

**Comment by jainaashish** [comment]
d928588849dd4fc32cd3f782cb584f8a8e9aa429

**Pull request closed without merging**